instrumentor/controllers/instrumentedapplication_controller.go

```diff
@@ -0,0 +1,362 @@
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodRunning {
+			return &pod, nil
+		}
+	}
+
+	return nil, consts.PodsNotFoundErr
+}
+
+func (r *InstrumentedApplicationReconciler) createLangDetectionPod(targetPod *corev1.Pod, instrumentedApp *v1.InstrumentedApplication) (*corev1.Pod, error) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-instrumentation-detection-", targetPod.Name),
+			Namespace:    targetPod.Namespace,
+			Annotations: map[string]string{
+				consts.InstrumentationDetectionContainerAnnotationKey: "true",
+				istioAnnotationKey:   istioAnnotationValue,
+				linkerdAnnotationKey: linkerdAnnotationValue,
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "instrumentation-detector",
+					Image: fmt.Sprintf("%s:%s", r.InstrumentationDetectorImage, r.InstrumentationDetectorTag),
+					Args: []string{
+						fmt.Sprintf("--pod-uid=%s", targetPod.UID),
+						fmt.Sprintf("--container-names=%s", strings.Join(r.getContainerNames(targetPod), ",")),
+					},
+					TerminationMessagePath: "/dev/detection-result",
+					SecurityContext: &corev1.SecurityContext{
+						Capabilities: &corev1.Capabilities{
+							Add: []corev1.Capability{"SYS_PTRACE"},
+						},
+					},
+				},
+			},
+			RestartPolicy: "Never",
+			NodeName:      targetPod.Spec.NodeName,
+			HostPID:       true,
+		},
+	}
+
+	err := ctrl.SetControllerReference(instrumentedApp, pod, r.Scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	return pod, nil
+}
+
+func (r *InstrumentedApplicationReconciler) getContainerNames(pod *corev1.Pod) []string {
+	var result []string
+	for _, c := range pod.Spec.Containers {
+		if !r.skipContainer(c.Name) {
+			result = append(result, c.Name)
+		}
+	}
+
+	return result
+}
+
+func (r *InstrumentedApplicationReconciler) skipContainer(name string) bool {
+	return name == "istio-proxy" || name == "linkerd-proxy"
+}
+
+func (r *InstrumentedApplicationReconciler) getOwnerTemplateLabels(ctx context.Context, instrumentedApp *v1.InstrumentedApplication) (map[string]string, error) {
+	owner := metav1.GetControllerOf(instrumentedApp)
+	if owner == nil {
+		return nil, errors.New("could not find owner for InstrumentedApp")
+	}
+
+	if owner.Kind == "Deployment" && owner.APIVersion == appsv1.SchemeGroupVersion.String() {
+		var dep appsv1.Deployment
+		err := r.Get(ctx, client.ObjectKey{
+			Namespace: instrumentedApp.Namespace,
+			Name:      owner.Name,
+		}, &dep)
+		if err != nil {
+			return nil, err
+		}
+
+		return dep.Spec.Template.Labels, nil
+	} else if owner.Kind == "StatefulSet" && owner.APIVersion == appsv1.SchemeGroupVersion.String() {
+		var ss appsv1.StatefulSet
+		err := r.Get(ctx, client.ObjectKey{
+			Namespace: instrumentedApp.Namespace,
+			Name:      owner.Name,
+		}, &ss)
+		if err != nil {
+			return nil, err
+		}
+
+		return ss.Spec.Template.Labels, nil
+	}
+
+	return nil, errors.New("unrecognized owner kind")
```

Comment:
Can we format this error to present which kind? And rephrase the message so that it will be like this, for example: "Daemonset is not supported"